### PR TITLE
aya-ebpf: Make use of `Borrow` and `BorrowMut` in map methods

### PR DESCRIPTION
**Pull request by vadorovsky**
Let callers pass either owned objects or references. We do that already in the user-space methods.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aya-rs/aya/1362)

### CODE REVIEW

File: test/integration-ebpf/src/uprobe_cookie.rs [hunk 1]
```diff
@@ -18,5 +18,5 @@ static RING_BUF: RingBuf = RingBuf::with_byte_size(0, 0);
 fn uprobe_cookie(ctx: ProbeContext) {
     let cookie = unsafe { helpers::bpf_get_attach_cookie(ctx.as_ptr()) };
     let cookie_bytes = cookie.to_le_bytes();
-    let _res = RING_BUF.output(&cookie_bytes, 0);
+    let _res = RING_BUF.output::<[u8; 8]>(&cookie_bytes, 0);
```
Comment (vadorovsky) [inline test/integration-ebpf/src/uprobe_cookie.rs:21]: This is an unfortunate outcome of this change - compiler requires this explicit annotation, because `[u8: 8]` can be borrowed either as `&[u8]` or `&[u8; 8]`. But this one edge case is still acceptable IMO.

Comment (tamird) [reply test/integration-ebpf/src/uprobe_cookie.rs:21]: Hmm. Could you pass by value here or pass &cookie_bytes[..]?

Comment (vadorovsky) [reply test/integration-ebpf/src/uprobe_cookie.rs:21]: Nope. I tried also `.as_slice()`, doesn't work. See https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=305b264b4a9d694af2c05eddb635bdb1

**Comment by vadorovsky** [comment]
@tamird Given that you approved and I haven't been able to get rid of the type annotation, I'm merging it.

File: test/integration-ebpf/src/uprobe_cookie.rs (hunk 1 above)
Comment (tamird) [reply test/integration-ebpf/src/uprobe_cookie.rs:21]: Ah, I see. I think the better place to put this annotation would be on RING_BUF.
